### PR TITLE
Install rosdeps with robot compile script before building pkgs

### PR DESCRIPTION
**Pull request by texhnolyze**
## Proposed changes
- fix execution of sudo commands in robot compile script by utilizing `ssh -t`
- rm davros from robot compile script
- auto install rosdeps before build, when an internet connection is available

## Necessary checks
- [x] Test on your machine
- [x] Test on the robot
- [x] Put the PR on our Project board

### CODE REVIEW

File: scripts/robot_compile.py [hunk 1]
```diff
@@ -379,23 +377,27 @@ def build(target, package='', pre_clean=False):
...
-def check_rosdeps(target):
+def install_rosdeps(target):
     """
-    Check installed dependencies on a target with rosdep
+    Install dependencies on a target with rosdep
 
     :type target: Target
     """
-    print_info("Checking installed rosdeps on {}".format(target.hostname))
+    if check_internet(target):
```
Comment (jaagut) [inline scripts/robot_compile.py:386]: I'd prefer `internet_available()`

File: scripts/robot_compile.py [hunk 2]
```diff
@@ -379,23 +377,27 @@ def build(target, package='', pre_clean=False):
...
+        print(cmd)
 
-    rosdep_result = _execute_on_target(target, cmd)
-    if rosdep_result.returncode != 0:
-        print_warn("rosdep check on {} had non-zero exit code. Check its output for more info".format(target.hostname))
-
-    print_success("Rosdeps on {} installed successfully".format(target.hostname))
+        rosdep_result = _execute_on_target(target, cmd)
+        if rosdep_result.returncode != 0:
+            print_warn(f"rosdep install on {target.hostname} had non-zero exit code. Check its output for more info")
```
Comment (jaagut) [inline scripts/robot_compile.py:396]:
```suggestion
            print_warn(f"Rosdep install on {target.hostname} had non-zero exit code. Check its output for more info")
```

File: scripts/robot_compile.py [hunk 3]
```diff
@@ -379,23 +377,27 @@ def build(target, package='', pre_clean=False):
...
 
+        print_success(f"rosdeps on {target.hostname} installed successfully")
```
Comment (jaagut) [inline scripts/robot_compile.py:398]:
```suggestion
        print_success(f"Rosdeps on {target.hostname} installed successfully")
```

File: scripts/robot_compile.py [hunk 4]
```diff
@@ -379,23 +377,27 @@ def build(target, package='', pre_clean=False):
...
+    else: 
+        print_info(f"skipping rosdep install on {target.hostname} as we do not have internet")
```
Comment (jaagut) [inline scripts/robot_compile.py:400]:
```suggestion
        print_info(f"Skipping rosdep install on {target.hostname} as we do not have internet")
```

**Review by jaagut — APPROVED** [review]

**Review by Flova — APPROVED** [review]

File: scripts/robot_compile.py [hunk 5]
```diff
@@ -379,23 +377,27 @@ def build(target, package='', pre_clean=False):
...
-    print_info("Checking installed rosdeps on {}".format(target.hostname))
+    if check_internet(target): 
+        print_info(f"Installing rosdeps on {target.hostname}")
+        target_src_path = os.path.join(target.workspace, "src")
+        extra_flags = "-q" if _should_run_quietly() else ""
 
-    cmd = "rosdep check {} --ignore-src --from-paths {}".format("" if LOGLEVEL.current >= LOGLEVEL.INFO else "-q",
-                                                                os.path.join(target.workspace, "src"))
+        cmd = f"rosdep install -y {extra_flags} --ignore-src --from-paths {target_src_path}"
+        print(cmd)
```
Comment (timonegk) [inline scripts/robot_compile.py:392]: Debug print?

**Review by jaagut — APPROVED** [review]

**Pull request merged**